astropy/table/ndarray_mixin.py

```diff
@@ -0,0 +1,66 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import numpy as np
+
+from astropy.utils.data_info import ParentDtypeInfo
+
+
+class NdarrayMixinInfo(ParentDtypeInfo):
+    _represent_as_dict_primary_data = 'data'
+
+    def _represent_as_dict(self):
+        """Represent Column as a dict that can be serialized."""
+        col = self._parent
+        out = {'data': col.view(np.ndarray)}
+        return out
+
+    def _construct_from_dict(self, map):
+        """Construct Column from ``map``."""
+        data = map.pop('data')
+        out = self._parent_cls(data, **map)
+        return out
+
+
+class NdarrayMixin(np.ndarray):
+    """
+    Mixin column class to allow storage of arbitrary numpy
+    ndarrays within a Table.  This is a subclass of numpy.ndarray
+    and has the same initialization options as ndarray().
+    """
+    __module__ = 'astropy.table.table'
+    info = NdarrayMixinInfo()
+
+    def __new__(cls, obj, *args, **kwargs):
+        self = np.array(obj, *args, **kwargs).view(cls)
+        if 'info' in getattr(obj, '__dict__', ()):
+            self.info = obj.info
+        return self
+
+    def __array_finalize__(self, obj):
+        if obj is None:
+            return
+
+        if callable(super().__array_finalize__):
+            super().__array_finalize__(obj)
```

Comment:
I don't really even know what this is doing or how to get this stage. We need the numpy-fu of @mhvk.

Reply:
This would be if a new class was created using the mixin somewhere in the middle. Very weirdly, `ndarray.__array_finalize__` is `None`, which means that one cannot just call super on it...  Anyway, I think it is fine to ignore the lack of coverage here...